ignite/metrics/nlp/bleu.py

```diff
@@ -187,8 +191,9 @@ def reset(self) -> None:
     @reinit__is_reduced
     def update(self, output: Tuple[Sequence[Any], Sequence[Sequence[Any]]]) -> None:
         y_pred, y = output
-        self._sum_of_bleu += self._corpus_bleu(references=[y], candidates=[y_pred])
-        self._num_sentences += 1
+        for _y_pred, _y in zip(y_pred, y):
+            self._sum_of_bleu += self._corpus_bleu(references=[_y], candidates=[_y_pred])
+            self._num_sentences += 1
```

Comment:
Looks like this breaks BC ?

Reply:
Yes it does. I was thinking since we don't have an official notebook, we could improve the usage to handle batches. What do you think ?

Reply:
I have an impression that type hints in `update` method are incorrect, especially checking the ones of `_corpus_bleu`

Reply:
If you want to keep compatibility with a warning, one way would be like this to add a check if single instances are passed like this https://github.com/pytorch/ignite/pull/2180#issuecomment-907169670

Reply:
OK, I understand now. Sorry, I haven't read everything here and now I agree with Sylvain and let's make the BC-breaking change. What about type hints, why they are the same as previous ones ?

Reply:
Okay, I'll update the type hints